File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> numsHash;
        vector<int> result;
        for (size_t index = 0; index < nums.size(); ++index) 
        {
            int lookingFor = target - nums[index];
            if (numsHash.find(lookingFor) != numsHash.end())
            {
                result.push_back(numsHash[lookingFor]);
                result.push_back(index);
                break;
            }
            numsHash[nums[index]] = index;
        }
        return result;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        start = 0
        end = start + 1
        indexes = []
        while start < len(nums) - 1:
            if(end == len(nums)):
                start += 1
                end = start + 1
            if nums[start] + nums[end] == target:
                indexes = [start, end]
                break
            else:
                end += 1
        return indexes